pkg/apis/planetscale/v2/vitesskeyspace_types.go

```diff
@@ -294,9 +294,36 @@ type VitessKeyspaceStatus struct {
 	// ReshardingInProgress indicates if there is a VReplication workflow in progress.
 	ReshardingInProgress corev1.ConditionStatus `json:"reshardingInProgress,omitempty"`
 	// ActiveWorkflows is a list of the current VReplication workflows in progress.
-	ActiveWorkflows []string `json:"activeWorkflows,omitempty"`
+	ActiveWorkflows []WorkflowStatus `json:"activeWorkflows,omitempty"`
+	// ServingShards is a list of shards that are currently serving writes.
+	ServingShards []string `json:"servingShards,omitempty"`
 }
 
+// WorkflowStatus defines some of the workflow related status information.
+type WorkflowStatus struct {
+	// Workflow represents the name of the workflow relevant to the related replication statuses.
+	Workflow string `json:"workflow"`
+	// State is either 'Running', 'Copying', or 'Lagging'.
+	State WorkflowState `json:"state"`
+	// UnsafeVReplicationLag indicates if VReplication lag is currently exceeding 10s for the given workflow
+	UnsafeVReplicationLag bool `json:"unsafeVReplicationLag,omitempty"`
+}
+
+// WorkflowState represents the current state for the given Workflow.
+type WorkflowState string
+
+const (
+	// RunningState indicates that the workflow is currently in the Running state. This state
+	// indicates that vreplication is ongoing, but we have moved passed the copying phase.
+	RunningState WorkflowState = "Running"
+	// CopyingState indicates that the workflow is currently in the Copying state.
+	CopyingState WorkflowState = "Copying"
+	// Lagging indicates that the workflow currently has vreplication lag exceeding 10 seconds.
```

Comment:
What's the difference between this `Lagging` state and the new transaction_timestamp stuff you added recently? From the Vitess code, it looks like the Lagging state only takes time_updated into account?

We should think about whether we need to expose both types of lag (time_updated vs. transaction_timestamp) in our status API. It seems confusing so it might be better to hide the complexity somehow.

For example, consumers currently need to know that Lagging is really a special case of Running (so a UI should still show that resharding is running, for example), and also that Lagging is a transient state that might revert to Running at any moment. Maybe we should just look at transaction_timestamp to report lag orthogonally to state, and hide this not-very-useful Lagging state by reporting Running at our status API level?

Reply:
These states mimick the four states that we will see in the vrepl row State results. @rohit-nayak-ps designed it such that the `State` field gets updated based on a set of criteria. `Lagging` already existed as a given state, but isn't as accurate as our new `MaxVReplicationLag` field. Yes, it only takes time_updated into account. 

This is mostly so we can translate from these known states. I guess another option is we can remove "Lagging" entirely and skip over it if we find it in a `ReplicationStatusResult`